src/Chapter1.hs

```diff
@@ -187,463 +184,460 @@ ghci> :q
-
-Casual reminder about adding top-level type signatures for all functions :)
--}
-
-mid x y z = error "mid: not implemented!"
-
-{- |
-=⚔️= Task 8
-
-Implement a function that checks whether a given character is a vowel.
-
-🕯 HINT: use guards
-
->>> isVowel 'a'
-True
->>> isVowel 'x'
-False
--}
-isVowel c = error "isVowel: not implemented!"
-
-
-{- |
-== Local variables and functions
-
-So far, we've been playing only with simple expressions and function
-definitions. However, in some cases, expressions may become complicated, and it
-could make sense to introduce some helper variables.
-
-You can use the let-in construction in Haskell to define variables.
-Here goes an example:
-
-@
-half :: Int -> Int
-half n = let halfN = div n 2 in halfN
-@
-
-♫ NOTE: __let-in__ is also an expression! You can't just define variables; you
-  also need to return some expression that may use defined variables.
-
-The syntax for defining multiple variables requires to care about indentation
-more, but there is nothing special in it as well:
-
-@
-halfAndTwice :: Int -> (Int, Int)
-halfAndTwice n =
-    let halfN = div n 2
-        twiceN = n * 2
-    in (halfN, twiceN)
-@
-
-In addition to let-in (or sometimes even alternatively to let-in) you can use
-the __where__ construction to define local variables and functions.
-And, again, the example:
-
-@
-pythagoras :: Double -> Double -> Double
-pythagoras a b = square a + square b
+closestToZero x y = if (abs x) < (abs y) then x else y
+
+-- |
+-- =⚔️= Task 7
+-- Write a function that returns the middle number among three given numbers.
+--
+-- >>> mid 3 1 2
+-- 2
+--
+-- 🕯 HINT: When checking multiple conditions, it is more convenient to use the
+--   language construction called "guards" instead of multiple nested 'if-then-else'
+--   expressions. The syntax of guards is the following:
+--
+-- @
+-- sign :: Int -> Int
+-- sign n
+--     | n < 0 = (-1)
+--     | n == 0 = 0
+--     | otherwise = 1
+-- @
+--
+-- You define different conditions in different branches, started by the '|'
+-- symbol. And the functions check them from top to bottom, returning the first
+-- value after "=" where the condition is true.
+--
+-- ♫ NOTE: The "=" sign goes after each branch, respectively.
+--
+-- ♫ NOTE: It is essential to have the same indentation before each branch "|"!
+--   Remember, that Haskell is indentation- and layout-sensitive language.
+--
+-- Casual reminder about adding top-level type signatures for all functions :)
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+  | y <= x && x <= z = x
+  | z <= x && x <= y = x
+  | x <= y && y <= z = y
+  | z <= y && y <= x = y
+  | otherwise = z
+
+-- |
+-- =⚔️= Task 8
+--
+-- Implement a function that checks whether a given character is a vowel.
+--
+-- 🕯 HINT: use guards
+--
+-- >>> isVowel 'a'
+-- True
+-- >>> isVowel 'x'
+-- False
+isVowel :: Char -> Bool
+isVowel 'a' = True
+isVowel 'e' = True
+isVowel 'i' = True
+isVowel 'o' = True
+isVowel 'u' = True
+isVowel _ = False
```

Comment:
Very nice usage of pattern-matching! :100: We explain it only in Chapter Two, but well done :+1: